src/modules/sync/utils/contentTypeHelpers.ts

```diff
@@ -1,41 +1,41 @@
 import { ContentTypeElements } from "@kontent-ai/management-sdk";
-import {
-  DomHtmlNode,
-  DomNode,
-  nodeParse,
-  ParseResult,
-  ResolveDomHtmlNodeType,
-  ResolveDomTextNodeType,
-  transformToJson,
-} from "@kontent-ai/rich-text-resolver";
+import { PortableTextObject, transformToPortableText } from "@kontent-ai/rich-text-resolver";
 
 import { notNullOrUndefined } from "../../../utils/typeguards.js";
 import { getAssetReferences, getItemReferences, OriginalReference } from "../diff/guidelinesRichText.js";
 import { SyncAssetElement, SyncLinkedItemsElement, SyncTypeElement } from "../types/syncModel.js";
 
-const resolveAssetIdsDomHtmlNode: ResolveDomHtmlNodeType = (node, traverse) => {
-  switch (node.tagName) {
-    case "figure":
-      return node.attributes["data-asset-id"];
-    case "a":
-      return node.attributes["data-asset-id"];
-    default: {
-      return node.children.map(traverse);
-    }
+const resolveAssetIdsDomHtmlNode = (node: PortableTextObject): ReadonlyArray<string> => {
+  if (node._type === "image") {
+    return node.asset.referenceType === "id" ? [node.asset._ref] : [];
+  }
+
+  if (node._type === "block") {
+    return node.markDefs?.filter(a => a._type === "link" && "data-asset-id" in a).map(a => (a as any)["data-asset-id"])
```

Comment:
Similarly can you use a different type here other than any?

Reply:
Sadly I think that for asset link the Type is missing :/